implementation.js

```diff
@@ -1,9 +1,88 @@
 'use strict';
 var ES = require('es-abstract/es6');
 var supportsDescriptors = require('define-properties').supportsDescriptors;
+var has = require('has');
+var global = require('global-object');
+
+var parseIterable = function (arrayLike) {
+	var done = false;
+	var iterableResponse;
+	var tempArray = [];
+
+	if (arrayLike && typeof arrayLike.next === 'function') {
+		while (!done) {
+			iterableResponse = arrayLike.next();
+			if (
+				has(iterableResponse, 'value')
+				&& has(iterableResponse, 'done')
+			) {
+				if (iterableResponse.done === true) {
+					done = true;
+					break; // eslint-disable-line no-restricted-syntax
+
+				} else if (iterableResponse.done !== false) {
+					break; // eslint-disable-line no-restricted-syntax
+				}
+
+				tempArray.push(iterableResponse.value);
+			} else if (iterableResponse.done === true) {
+				done = true;
+				break; // eslint-disable-line no-restricted-syntax
+			} else {
+				break; // eslint-disable-line no-restricted-syntax
+			}
+		}
+	}
+
+	return done ? tempArray : false;
+};
+
+var hasSymbols = typeof Symbol === 'function' && typeof Symbol.iterator === 'symbol';
+var iteratorSymbol;
+if (hasSymbols) {
+	iteratorSymbol = Symbol.iterator;
+} else if ('Set' in global) {
+	var supportsStrIterator = (function () {
+		try {
+			return eval( // eslint-disable-line no-eval
+				'var nativelySupportsStringIterator = false;'
+				+ 'var s = new Set();'
+				+ 's.add(1);'
+				+ 's.add(2);'
+				+ 's["@@iterator"] = function () {'
+					+ 'nativelySupportsStringIterator = true;'
+				+ '};'
+				+ 'for (var x of s) {}'
+				+ 'return nativelySupportsStringIterator;'
+			);
+		} catch (e) {
+			return false;
+		}
+	}());
+
+	if (supportsStrIterator) {
+		iteratorSymbol = '@@iterator';
+	}
+}
+
+var isCallable = require('is-callable');
+
+var usingIterator = function (items) {
+	if (iteratorSymbol && isCallable(items[iteratorSymbol])) {
+		return items[iteratorSymbol]();
+	} else if ('Set' in global && 'Map' in global && isCallable(items.entries) && isCallable(items.values)) {
+		if (items.constructor === 'Set') {
```

Comment:
wouldn't the `constructor` property `=== Set`, not the *string* "Set"? (same with Map below)